### PR TITLE
Add some unit tests for the security client.

### PR DESCRIPTION
**Pull request by anthonyvercolano**
# Description of the problem

More unit tests are needed for the security client.
# Description of the solution
 
This adds some unit tests for the security client.  Still have a way to go.

### CODE REVIEW

File: provisioning/device/samples/register_tpm.js [hunk 1]
```diff
@@ -29,6 +29,7 @@ var idScope = '<the id scope for your dps instance>';
 
 var securityClient = new tpmSecurity.TpmSecurityClient();
 
+var provisioningClient = ProvisioningDeviceClient.create(provisioningHost, idScope, new provisioningTransport(), securityClient);
```
Comment (pierreca) [inline provisioning/device/samples/register_tpm.js]: nit (fix later): constructors should be pascal-cased (ProvisioningTransport)

File: security/tpm/src/tpm.ts [hunk 2]
```diff
@@ -288,8 +285,28 @@ export class TpmSecurityClient  {
...
+    });
+  }
+
+
   private _signData(dataToSign: Buffer, callback: (err: Error, signedData: Buffer) => void): void {
 
+    /*Codes_SRS_NODE_TPM_SECURITY_CLIENT_06_013: [If `signWithIdentity` is invoked without a previous successful invocation of `activateIdentityKey`, the callback will be invoked with `err` of `InvalidOperationError`] */
+    if (!this._idKeyPub) {
+      callback(new errors.InvalidOperationError('activateIdentityKey must be invoked before any signing is attempted.'), null);
+      return;
```
Comment (pierreca) [inline security/tpm/src/tpm.ts]: nit: the `,null` is unnecessary and you could in theory just `return callback(new errors...)` that way it doesn't look like something happens after the callback (even if it's just a return)

File: security/tpm/test/_tpm_test.js [hunk 3]
```diff
@@ -97,11 +100,30 @@ describe('tpm', function () {
       testFalsyArg('signWithIdentity', 'dataToSign', falsyDataToSign, ReferenceError);
     });
 
-    it('must call activateIdentityKey first', function() {
-      /*Tests_SRS_NODE_TPM_SECURITY_CLIENT_06_013: [If `signWithIdentity` is invoked without a previous successful invocation of `activateIdentityKey`, an InvalidOperationError is thrown.] */
+    it('must call activateIdentityKey first', function(done) {
+      /*Tests_SRS_NODE_TPM_SECURITY_CLIENT_06_013: [If `signWithIdentity` is invoked without a previous successful invocation of `activateIdentityKey`, the callback will be invoked with `err` of `InvalidOperationError`.] */
```
Comment (pierreca) [inline security/tpm/test/_tpm_test.js]: so I guess "previous successful invocation" now applies to the lifecycle of the TPM but not the TPM security client state machine itself, right?

File: provisioning/device/samples/register_tpm.js (hunk 1 above)
Comment (anthonyvercolano) [reply provisioning/device/samples/register_tpm.js]: Indeed.

File: security/tpm/src/tpm.ts (hunk 2 above)
Comment (anthonyvercolano) [reply security/tpm/src/tpm.ts]: Well as it was written, the callback required signedData to be returned.  Added the ?

File: security/tpm/test/_tpm_test.js (hunk 3 above)
Comment (anthonyvercolano) [reply security/tpm/test/_tpm_test.js]: Yes, somewhere in the depths of time, activateIdentityKey needs to be called.

**Pull request merged**